### PR TITLE
Add `BoundedInt` and internal macros, update cairo

### LINKED ISSUE #597: Use BoundedInt for max_u256
When [this commit](https://github.com/starkware-libs/cairo/commit/e1165f133db4af0e8f3765c8853f3b86e8e50e58) makes it into a release of cairo, then [these lines](https://github.com/OpenZeppelin/cairo-contracts/blob/a54e98ca74e0d2c127680f3a3ca23d4643eb0ddf/src/openzeppelin/token/erc20.cairo#L198-L206) should be rewritten as
```rust
    fn _spend_allowance(owner: ContractAddress, spender: ContractAddress, amount: u256) {
        let current_allowance = _allowances::read((owner, spender));
        if current_allowance == BoundedInt::max() {
            _approve(owner, spender, current_allowance - amount);
        }
    }
```

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes #597.

#### PR Checklist

- [x] Tests
- [ ] Tried the feature on a public network
- [ ] Documentation

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,7 @@ build:
 	cargo build
 
 test:
-	cargo run --bin cairo-test -- --starknet --path $(SOURCE_FOLDER)
+	cargo run --bin cairo-test -- --starknet $(SOURCE_FOLDER)
```
Comment (martriay) [inline Makefile:15]: is this a `1.0.0-alpha.7` change?

File: src/openzeppelin/tests/test_erc20.cairo [hunk 2]
```diff
@@ -13,9 +14,7 @@ const NAME: felt252 = 111;
 const SYMBOL: felt252 = 222;
 
 fn MAX_U256() -> u256 {
```
Comment (martriay) [inline src/openzeppelin/tests/test_erc20.cairo:16]: do we need this function at all now?

**Review by martriay — COMMENTED** [review]
much better :)

please merge `cairo-1` as soon as we merge https://github.com/OpenZeppelin/cairo-contracts/pull/613 so we can have the CI

File: src/openzeppelin/tests/test_erc20.cairo (hunk 2 above)
Comment (andrew-fleming) [reply src/openzeppelin/tests/test_erc20.cairo:16]: Nope, good call

File: Makefile (hunk 1 above)
Comment (andrew-fleming) [reply Makefile:15]: Post alpha 7: https://github.com/starkware-libs/cairo/commit/3f4da6c0c9aa1672da4be73f10b6e386480c0455

File: src/openzeppelin/tests/test_erc20.cairo [hunk 3]
```diff
@@ -399,12 +394,12 @@ fn test__spend_allowance_unlimited() {
     let (owner, supply) = setup();
 
     let spender: ContractAddress = contract_address_const::<2>();
-    let max_minus_one: u256 = MAX_U256() - u256_from_felt252(1);
+    let max_minus_one: u256 = BoundedInt::max() - u256_from_felt252(1);
```
Comment (Amxx) [inline src/openzeppelin/tests/test_erc20.cairo:397]:
```suggestion
    let max_minus_one: u256 = BoundedInt::max() - 1.into();
```

**Review by martriay — APPROVED** [review]

**Review by martriay — APPROVED** [review]

**Review by martriay — APPROVED** [review]

**Pull request merged**